pkg/encryption/path_test.go

```diff
@@ -98,3 +100,60 @@ func forAllCiphers(test func(cipher storj.CipherSuite)) {
 		test(cipher)
 	}
 }
+
+func TestSegmentEncoding(t *testing.T) {
+	segments := [][]byte{
+		[]byte{},
+		[]byte{'a'},
+		[]byte{0},
+		[]byte{'/'},
+		[]byte{'a', 'b', 'c', 'd', '1', '2', '3', '4', '5'},
+
+		[]byte{'/', '/', '/', '/', '/'},
+		[]byte{0, 0, 0, 0, 0, 0, 0, 0, 0},
+
+		[]byte{'a', '/', 'a', '2', 'a', 'a', 0, '1', 'b', 255},
+		[]byte{'/', '/', 'a', 0, 'a', 'a', 0, '1', 'b', 'g', 'a', 'b', '/'},
+		[]byte{0, '/', 'a', '0', 'a', 'a', 0, '1', 'b', 'g', 'a', 'b', 0},
+	}
+
+	for _, segment := range segments {
+		encoded := encodeSegment(segment)
```

Comment:
Can we also require that `bytes.IndexByte(encoded, {0, 255, '/'})` is always `-1`?

Reply:
done